src/tss2-esys/esys_iutil.c

```diff
@@ -387,6 +387,11 @@ iesys_handle_to_tpm_handle(ESYS_TR esys_handle, TPM2_HANDLE * tpm_handle)
         *tpm_handle = TPM2_RH_ACT_0 + (esys_handle - ESYS_TR_RH_ACT_FIRST);
         return TPM2_RC_SUCCESS;
     }
+    if (esys_handle >= ESYS_TR_RH_AC_FIRST &&
+        esys_handle <= ESYS_TR_RH_AC_LAST) {
+        *tpm_handle = TPM2_RH_ACT_0 + (esys_handle - ESYS_TR_RH_AC_FIRST);
```

Comment:
`ESYS_TR_RH_AC_FIRST` should map to `TPM2_NV_AC_FIRST` and `ESYS_TR_RH_AC_LAST` should map to `TPM2_NV_AC_FIRST`, so I think this ( think you just accidentally left `TPM2_RH_ACT_0`):

```C
if (esys_handle >= ESYS_TR_RH_AC_FIRST &&
        esys_handle <= ESYS_TR_RH_AC_LAST) {
    * tpm_handle = TPM2_NV_AC_FIRST + (esy_handle - ESYS_TR_RH_AC_FIRST);
}
```